apps/react/src/store/studio/slice.ts

```diff
@@ -0,0 +1,47 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { fetchStudios } from './dispatchers';
+import { initialState } from './state';
+
+/** Slice of the Redux store for managing studios data. */
+export const studiosSlice = createSlice({
+	name: 'studios',
+	initialState,
+	reducers: {
+		updateCursor(state) {
+			state.cursor = state.nextCursor;
+		},
+		resetCursor(state) {
+			state.cursor = '';
+		},
+		setPaginationEvent(state, action) {
+			state.isPaginationEvent = action.payload;
+		},
+		setSearchValue(state, action) {
+			state.searchValue = action.payload;
+		},
+		setOrdering(state, action) {
+			state.ordering = action.payload;
+		},
+	},
+	extraReducers: builder => builder
+		.addCase(fetchStudios.pending, state => {
+			state.isLoading = true;
+		})
+		.addCase(fetchStudios.fulfilled, (state, action) => {
+			const { results, nextCursor } = action.payload;
+			state.studios = state.isPaginationEvent ? [...state.studios, ...results] : [...results];
```

Comment:
Prefer to use `concat` in such cases:
```suggestion
			state.studios = state.isPaginationEvent ? state.studios.concat(results) : [...results];
```